pennylane/math/fn.py

```diff
@@ -163,6 +163,33 @@ def arcsin(tensor):
     return TensorBox(tensor).arcsin(wrap_output=False)
 
 
+def block_diag(values):
+    """Combine a sequence of 2D tensors to form a block diagonal tensor.
+
+    Args:
+        values (Sequence[tensor_like]): Sequence of 2D arrays/tensors to form
+            the block diagonal tensor.
+
+    Returns:
+        tensor_like: the block diagonal tensor
+
+    **Example**
+
+    >>> t = [
+    ...     np.array([[1, 2], [3, 4]]),
+    ...     torch.tensor([[1, 2, 3], [-1, -6, -3]]),
+    ...     torch.tensor(5)
+    ... ]
+    >>> qml.math.block_diag(t)
+    tensor([[ 1,  2,  0,  0,  0,  0],
+        [ 3,  4,  0,  0,  0,  0],
+        [ 0,  0,  1,  2,  3,  0],
+        [ 0,  0, -1, -6, -3,  0],
+        [ 0,  0,  0,  0,  0,  5]])

```

Comment:
Minor one, but will this format correctly?

Reply:
Fixed! Thanks for catching this one